src/main/java/org/opensearch/ad/transport/IndexAnomalyDetectorTransportAction.java

```diff
@@ -99,8 +107,34 @@ protected void doExecute(Task task, IndexAnomalyDetectorRequest request, ActionL
         RestRequest.Method method = request.getMethod();
         String errorMessage = method == RestRequest.Method.PUT ? FAIL_TO_UPDATE_DETECTOR : FAIL_TO_CREATE_DETECTOR;
         ActionListener<IndexAnomalyDetectorResponse> listener = wrapRestActionListener(actionListener, errorMessage);
+
+        Version nodeVersion = transportService.getLocalNode().getVersion();
```

Comment:
Can use the nodeVersion defined in the constructor?

Reply:
I need to remove that. There is a possibility that the one initialized in constructor may be null.